src/vs/editor/common/model/textModel.ts

```diff
@@ -1790,6 +1793,7 @@ export class TextModel extends Disposable implements model.ITextModel {
 
 		this._emitModelTokensChangedEvent({
 			tokenizationSupportChanged: false,
+			semanticTokensApplied: true,
```

Comment:
This is a bit too eager to set `semanticTokensApplied: true`. I would suggest to leave this to false to avoid a false positive, where some piece of code calls `clearSemanticTokens` even if real semantic tokens were never set.

Reply:
oh, right, actually I wanted to set this false, and only set semanticTokensApplied to true when there are actual tokens to set